### PR TITLE
Run LocalCommandLineCodeExecutor within venv

### PR DESCRIPTION
**Pull request by gziz**
## Why are these changes needed?
In AutoGen version 0.4, the current `LocalCommandLineCodeExecutor` doesn’t provide the feature to create a virtual environment on the fly for an application. This feature has been requested in https://github.com/microsoft/autogen/issues/2613.

This PR includes:
- Modifications to `LocalCommandLineCodeExecutor` to add this functionality.
- A function to create the virtual environment.
- Relevant tests for the new functionality and environment creation.
- Updates to the documentation of `LocalCommandLineCodeExecutor` to reflect this new feature.

Note: I based most of these changes on the current  AutoGen 0.2 equivalent code.

## Related issue number
https://github.com/microsoft/autogen/issues/2613

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

## Others
Here's an image showing the newly added documentation
Documentation Image

### CODE REVIEW

File: python/packages/autogen-core/src/autogen_core/components/code_executor/_impl/utils.py [hunk 1]
```diff
@@ -104,3 +106,46 @@ def infer_lang(code: str) -> str:
     except SyntaxError:
         # not a valid python code
         return "unknown"
+
+
```
Comment (gziz) [inline python/packages/autogen-core/src/autogen_core/components/code_executor/_impl/utils.py:110]: I have to say `create_virtual_environment` ended up being long, it's possible users would want to specify the parameters. However, if desired, we could modify this function to just take in `dir_path` and leave the rest of the values as defaults.

Comment (ekzhu) [reply python/packages/autogen-core/src/autogen_core/components/code_executor/_impl/utils.py:110]: I agree with this function being too long. Looking under the hood it is just a wrapper around `venv.EnvBuilder`, I suggest we remove this function and document the usage of `venv.EnvBuilder` directly

Comment (gziz) [reply python/packages/autogen-core/src/autogen_core/components/code_executor/_impl/utils.py:110]: Good idea!

File: python/packages/autogen-core/src/autogen_core/components/code_executor/_impl/local_commandline_code_executor.py [hunk 2]
```diff
@@ -54,6 +56,7 @@ class LocalCommandLineCodeExecutor(CodeExecutor):
             directory is the current directory ".".
         functions (List[Union[FunctionWithRequirements[Any, A], Callable[..., Any]]]): A list of functions that are available to the code executor. Default is an empty list.
         functions_module (str, optional): The name of the module that will be created to store the functions. Defaults to "functions".
+        virtual_env_context (Optional[SimpleNamespace], optional): The virtual environment context. Defaults to None.
```
[the remark this answers is not in the document]
Comment (gziz) [reply python/packages/autogen-core/src/autogen_core/components/code_executor/_impl/local_commandline_code_executor.py:59]: image

File: python/packages/autogen-core/src/autogen_core/components/code_executor/_impl/local_commandline_code_executor.py [hunk 3]
```diff
@@ -54,6 +56,33 @@ class LocalCommandLineCodeExecutor(CodeExecutor):
...
+
+    Example:
```
Comment (ekzhu) [inline python/packages/autogen-core/src/autogen_core/components/code_executor/_impl/local_commandline_code_executor.py:61]: Some explanation on what the following example code block does.

**Review by ekzhu — APPROVED** [review]

**Pull request merged**